pinot-core/src/main/java/org/apache/pinot/core/operator/query/SelectionOrderByOperator.java

```diff
@@ -127,14 +125,22 @@ private Comparator<Object[]> getComparator() {
     // Compare all single-value columns
     int numOrderByExpressions = _orderByExpressions.size();
     List<Integer> valueIndexList = new ArrayList<>(numOrderByExpressions);
+    List<OrderByExpressionContext> mvOrderByExpressions = new ArrayList<>();
     for (int i = 0; i < numOrderByExpressions; i++) {
       if (_orderByExpressionMetadata[i].isSingleValue()) {
         valueIndexList.add(i);
       } else {
-        _queryHasMVSelectionOrderBy = true;
+        mvOrderByExpressions.add(_orderByExpressions.get(i));
       }
     }
 
+    if (mvOrderByExpressions.size() > 0) {
+      // MV columns should not be part of the selection order by only list
+      throw new UnsupportedOperationException(
```

Comment:
This should be `BadQueryRequestException`, so that we can gracefully handle it if necessary.